gbdk-lib/examples/cross-platform/platformer_template/src/player.c

```diff
@@ -0,0 +1,355 @@
+#include "camera.h"
+#include "level.h"
+
+BANKREF_EXTERN(PlayerCharacterLeft)
+BANKREF_EXTERN(PlayerCharacterRight)
+
+
+#define GRAVTY 15
+#define GROUND_FRICTION 15
+#define PLAYER_CHARACTER_INCREASE_JUMP_TIMER_MAX 20
+#define PLAYER_CHARACTER_JUMP_VELOCITY 450
+#define PLAYER_CHARACTER_WALK_VELOCITY 225
+#define PLAYER_CHARACTER_RUN_VELOCITY 325
+#define PLAYER_CHARACTER_WALK_TWO_FRAME_COUNTER 3
+#define PLAYER_CHARACTER_RUN_TWO_FRAME_COUNTER 5
+
+#define PLAYER_CHARACTER_METASPRITE_PIVOT_X 12
+#define PLAYER_CHARACTER_METASPRITE_PIVOT_Y 6
+#define PLAYER_CHARACTER_BOUNDING_BOX_HALF_WIDTH 5
+#define PLAYER_CHARACTER_BOUNDING_BOX_HALF_HEIGHT 12
+#define PLAYER_CHARACTER_BOUNDING_BOX_HEIGHT 24
+
+uint8_t facingRight =TRUE;
+
+uint8_t playerJumpIncrease = 0;
+uint8_t threeFrameCounter=0;
+uint16_t playerX, playerY;
+int16_t playerXVelocity, playerYVelocity;
+
+    // Flip horizontally, if we aren't facing right
+
+#if defined(NINTENDO_NES)
+    const uint8_t baseProp=4;
+#else
+    const uint8_t baseProp=0;
+#endif
+
+#if defined(SEGA)
+    #define PLAYER_PALETTES_BANK CURRENT_BANK
+    #define PLAYER_PALETTES PlayerPalettesGGSMS
+    #define set_player_sprite_data set_sprite_native_data
+
+#else
+    #define PLAYER_PALETTES_BANK BANK(PlayerCharacterLeft)
+    #define PLAYER_PALETTES PlayerCharacterLeft_palettes
+    #define set_player_sprite_data set_sprite_data
+#endif
+
+
+
+const palette_color_t PlayerPalettesGGSMS[16] = {
+	RGB8(255,128, 64), RGB8(248,248,248), RGB8(168,168,168), RGB8(  0,  0,  0)
+	,
+	RGB8(143,  0,  0), RGB8(  6,112,  0), RGB8(  0, 31,173), RGB8(122,134,  0)
+	,
+	RGB8(  0,138,111), RGB8( 75,  0, 82), RGB8(255,  0,  0), RGB8( 33,255,  0)
+	,
+	RGB8(  0, 46,255), RGB8(255,238,  0), RGB8(  0,225,255), RGB8(253,132,255)
+	
+};
+
+/**
+ * @brief We'll Put the PlayerCharacter's tiles in VRAM.
+ * Not every platform supports tile-flipping. We want To avoid wasting vram space with both left & right animations.
+ * we'll keep tiles only for the direction we are facing.
+ */
+void UpdatePlayerVRAMTiles() NONBANKED{
+    uint8_t _previous_bank = CURRENT_BANK;
+
+    if(facingRight){
+
+        SWITCH_ROM(BANK(PlayerCharacterRight));
+
+        set_player_sprite_data (0,PlayerCharacterRight_TILE_COUNT,PlayerCharacterRight_tiles);
+    } else {
+
+        SWITCH_ROM(BANK(PlayerCharacterLeft));
+
+        set_player_sprite_data (0,PlayerCharacterLeft_TILE_COUNT,PlayerCharacterLeft_tiles);
+    }
+    SWITCH_ROM(_previous_bank);
+}
+
+void SetPlayerPalettes() NONBANKED{
+    uint8_t _previous_bank = CURRENT_BANK;
+
+        SWITCH_ROM(PLAYER_PALETTES_BANK);
+    
+    // Set up color palettes
+    #if defined(SEGA)
+            set_sprite_palette(baseProp, 1, PLAYER_PALETTES);
+    #elif defined(GAMEBOY)
+        if (_cpu == CGB_TYPE) {
+            set_sprite_palette(OAMF_CGB_PAL0, 1, PLAYER_PALETTES);
+        }
+    #elif defined(NINTENDO_NES)
+        set_sprite_palette(baseProp, 4, PLAYER_PALETTES);
```

Comment:
I don't think using baseProp here is very useful, as it has nothing to do with the flipping described earlier.

As you already have separate if-clauses for each system, I would just have the first parameter be the correct value there. It will make things more readable.

As a side-note, I think we'll need to have a review/discussion about how to handle the palette indices for CGB vs NES. They are a little bit inconsistent at the moment. (but that can be done after merging this)